### DISCUSSION TITLE
What's the Best Way to Implement the Wazuh Agent into a Proxmox Template?

### DISCUSSION THREAD

**Discussion by echumley** [wazuh/wazuh#19710] [General]
Hello!

I've just starting using Wazuh and so far I'm loving it. I'd like to implement the Wazuh agent into my Proxmox VM and LXC templates so anytime I deploy a new VM or LXC, they'll automatically populate in the dashboard as new agents. Would there be any issue with doing doing that with how the agent runs right now or is there a 'proper' way of doing this? I know many services like this lock onto something like the host name or the machine ID of the VM and use that to differential the different agent installations, but my Machine IDs and host names change with every instance I clone.

Is there something in the back end preventing me from doing this? Thanks!

**Comment by wchesley** [comment]
There is no clean way to bake the agent into a PVE template as you can't change the agent name post-deployment. 

I use ansible to deploy agents after cloning a template.
